components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -13,292 +11,241 @@ import ChannelInviteModal from 'components/channel_invite_modal';
-                />
-            );
-        }
+    let markAsReadMenuItem: JSX.Element | null = null;
+    if (props.isUnread) {
+        markAsReadMenuItem = (
+            <Menu.ItemAction
+                id={`markAsRead-${props.channel.id}`}
+                onClick={handleMarkAsRead}
+                icon={<i className='icon-mark-as-unread'/>}
+                text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsRead', defaultMessage: 'Mark as Read'})}
+            />
+        );
+    }
 
-        let copyLink;
-        if (channel.type === Constants.OPEN_CHANNEL || channel.type === Constants.PRIVATE_CHANNEL) {
-            copyLink = (
-                <Menu.ItemAction
-                    id={`copyLink-${channel.id}`}
-                    onClick={this.copyLink}
-                    icon={<i className='icon-link-variant'/>}
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.copyLink', defaultMessage: 'Copy Link'})}
-                />
-            );
-        }
+    let markAsUnreadMenuItem: JSX.Element | null = null;
+    if (!props.isUnread) {
+        markAsUnreadMenuItem = (
+            <Menu.ItemAction
+                id={`markAsUnread-${props.channel.id}`}
+                onClick={handleMarkAsUnread}
+                icon={<i className='icon-mark-as-unread'/>}
+                text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsUnread', defaultMessage: 'Mark as Unread'})}
+            />
+        );
+    }
 
-        let addMembers;
-        if ((channel.type === Constants.PRIVATE_CHANNEL && this.props.managePrivateChannelMembers) || (channel.type === Constants.OPEN_CHANNEL && this.props.managePublicChannelMembers)) {
-            addMembers = (
-                <Menu.ItemAction
-                    id={`addMembers-${channel.id}`}
-                    onClick={this.addMembers}
-                    icon={<i className='icon-account-outline'/>}
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.addMembers', defaultMessage: 'Add Members'})}
-                />
-            );
-        }
+    let favoriteMenuItem: JSX.Element | null = null;
+    if (props.isFavorite) {
+        favoriteMenuItem = (
+            <Menu.ItemAction
+                id={`unfavorite-${props.channel.id}`}
+                onClick={handleUnfavoriteChannel}
+                icon={<i className='icon-star'/>}
+                text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.unfavoriteChannel', defaultMessage: 'Unfavorite'})}
+            />
+        );
+    } else {
+        favoriteMenuItem = (
+            <Menu.ItemAction
+                id={`favorite-${props.channel.id}`}
+                onClick={handleFavoriteChannel}
+                icon={<i className='icon-star-outline'/>}
+                text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.favoriteChannel', defaultMessage: 'Favorite'})}
+            />
```

Comment:
@larkox would it be overkill to have these behind useMemo, or are we ok to re-render these small components again?

Reply:
These are components, and they don't require much process to create them. When this is handled by React, it should be smart enough to do all the needed memoizations, so extra memoization here is not needed.

That being said, this pattern makes the file huge and a bit hard to read/review. I wonder if it makes sense to separate each item action into a component, but 0/5.